ui/page/components/utils.go

```diff
@@ -12,3 +17,22 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// RetryFunc implement retry policy for processes that needs to be executed
+// in the background after initial failure.
+func RetryFunc(retryAttempts int, sleepDur time.Duration, errFunc func() error) (int, error) {
+	var err error
+	for i := 0; i < retryAttempts; i++ {
+		if i > 0 {
+			sleepDur *= 2
+			log.Printf("waiting %v to retry after error: %v", sleepDur, err)
```

Comment:
To retry what? This wouldn't be a very helpful log message. A previous review comment recommended adding a `funcDesc` parameter to this function which can be included in this log for better clarity.